requirements.txt

```diff
@@ -1,3 +1,3 @@
 docrep>=0.3.1
-numpy>=1.19.0
+numpy>=1.19.0,<1.25.0
```

Comment:
I think this is too restrictive; based on your comment, this is hardware related (works on my M1), would rather skip the test for certain architectures and investigate further.

Reply:
It is OS related I would say (not hardware related): with MacOS the test never fails but with Linux...
Or did you try it with Linux?

Reply:
(M1 to M5 are the test matrices used... nothing to do with hardware...)

Reply:
Yes, sorry, just went too quickly through this PR. Seems really like a problem with the combination `numpy>=1.25.0` and `ubuntu`, `numpy==1.24.4` on `ubuntu` seems to work well (based on the 3.8 job).
Would consider skipping the combination using this combination.

Reply:
I looked at the `1.25.0` release notes, it wasn't immediately obvious which change would be affecting this.

Reply:
Actually, `1.25.0` was OK here: https://github.com/msmdev/pyGPCCA/actions/runs/5306789772/job/14363166673

Reply:
The only thing that looked suspicious to me when going through the numpy changelog was the change in `np.r_[]` and `np.c_[]`. We are using these in the troubled test.

Reply:
It is indeed strange that numpy 1.25.0 worked with python 3.9 on ubuntu one time but on the other hand pinning numpy below 1.25.0 solved the problem in all cases. So there should be a connection...

Reply:
Taking a closer look, I recognized that in the case were 1.25.0 worked, was the test using slepc, while 1.25.0 never succeeded without using slepc. This makes some sense to me...